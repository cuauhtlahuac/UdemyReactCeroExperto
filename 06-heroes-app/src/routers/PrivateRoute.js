import React from 'react';
import PropTypes from 'prop-types';
import { Redirect, Route } from 'react-router-dom';

const PrivateRoute = ({ isAuthenticated, component: Component, ...rest }) => {
	localStorage.setItem('lastPath', rest.location.pathname);

	const handleAuth = props => {
		if (isAuthenticated) {
			return <Component {...props} />;
		}
		return <Redirect to="/login" />;
	};

	return <Route {...rest} component={handleAuth} />;
};

PrivateRoute.propTypes = {
	isAuthenticated: PropTypes.bool.isRequired,
	component: PropTypes.func.isRequired,
	rest: PropTypes.any,
};

export default PrivateRoute;
