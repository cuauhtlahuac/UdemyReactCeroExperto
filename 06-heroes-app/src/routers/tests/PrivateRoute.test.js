import { shallow } from 'enzyme';
import React from 'react';
import PrivateRoute from 'routers/PrivateRoute';

describe('Pruebas de <PrivateRoute />', () => {
  let props = {};
  let component = () => <p>x</p>;

	beforeEach(() => {
		props = {
			isAuthenticated: false,
			component: jest.fn(() => <p>dokdokd</p>),
			location: { pathname: '/' },
		};
    component = shallow(<PrivateRoute {...props} />);
	});

	test('should render correctly', () => {

    expect(component).toMatchSnapshot();
	});
  
	test('should show component if isAuthenticated and save in localStorage', () => {
    component.setProps({...props, isAuthenticated: true })
    
    expect(component).toMatchSnapshot();
  });
});
