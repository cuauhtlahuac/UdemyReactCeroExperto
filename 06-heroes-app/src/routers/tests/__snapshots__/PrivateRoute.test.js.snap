// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas de <PrivateRoute /> should render correctly 1`] = `
<Route
  component={[Function]}
  location={
    Object {
      "pathname": "/",
    }
  }
/>
`;

exports[`Pruebas de <PrivateRoute /> should show component if isAuthenticated and save in localStorage 1`] = `
<Route
  component={[Function]}
  location={
    Object {
      "pathname": "/",
    }
  }
/>
`;
